tests/components/netatmo/test_camera.py

```diff
@@ -0,0 +1,185 @@
+
+
+async def test_camera_image_local(hass, camera_entry, requests_mock):
+    """Test retrieval or local camera image."""
+    await hass.async_block_till_done()
+
+    uri = "http://192.168.0.123/678460a0d47e5618699fb31169e2b47d"
+    stream_uri = uri + "/live/files/high/index.m3u8"
+    camera_entity_indoor = "camera.netatmo_hall"
+    cam = hass.states.get(camera_entity_indoor)
+
+    assert cam is not None
+    assert cam.state == STATE_STREAMING
+
+    stream_source = await camera.async_get_stream_source(hass, camera_entity_indoor)
+    assert stream_source == stream_uri
+
+    requests_mock.get(
+        uri + "/live/snapshot_720.jpg",
+        content=IMAGE_BYTES_FROM_STREAM,
+    )
+    image = await camera.async_get_image(hass, camera_entity_indoor)
+    assert image.content == IMAGE_BYTES_FROM_STREAM
+
+
+async def test_camera_image_vpn(hass, camera_entry, requests_mock):
+    """Test retrieval or remote camera image."""
+    await hass.async_block_till_done()
+
+    uri = (
+        "https://prodvpn-eu-2.netatmo.net/restricted/10.255.248.91/"
+        "6d278460699e56180d47ab47169efb31/MpEylTU2MDYzNjRVD-LJxUnIndumKzLboeAwMDqTTw,,"
+    )
+    stream_uri = uri + "/live/files/high/index.m3u8"
+    camera_entity_indoor = "camera.netatmo_garden"
+    cam = hass.states.get(camera_entity_indoor)
+
+    assert cam is not None
+    assert cam.state == STATE_STREAMING
+
+    stream_source = await camera.async_get_stream_source(hass, camera_entity_indoor)
+    assert stream_source == stream_uri
+
+    requests_mock.get(
+        uri + "/live/snapshot_720.jpg",
+        content=IMAGE_BYTES_FROM_STREAM,
+    )
+    image = await camera.async_get_image(hass, camera_entity_indoor)
+    assert image.content == IMAGE_BYTES_FROM_STREAM
+
+
+async def test_service_set_person_away(hass, camera_entry, caplog):
+    """Test service to set person as away."""
+    await hass.async_block_till_done()
+
+    data = {
+        "entity_id": "camera.netatmo_hall",
+        "person": "Richard Doe",
+    }
+
+    await hass.services.async_call(
+        "netatmo", SERVICE_SET_PERSON_AWAY, service_data=data
+    )
+    await hass.async_block_till_done()
+
+    assert "Set Richard Doe as away" in caplog.text
+
+    data = {
+        "entity_id": "camera.netatmo_hall",
+    }
+
+    await hass.services.async_call(
+        "netatmo", SERVICE_SET_PERSON_AWAY, service_data=data
+    )
+    await hass.async_block_till_done()
+
+    assert "Set home as empty" in caplog.text
+
+
+async def test_service_set_persons_home(hass, camera_entry, caplog):
+    """Test service to set persons as home."""
+    await hass.async_block_till_done()
+
+    data = {
+        "entity_id": "camera.netatmo_hall",
+        "persons": "John Doe",
+    }
+
+    await hass.services.async_call(
+        "netatmo", SERVICE_SET_PERSONS_HOME, service_data=data
+    )
+    await hass.async_block_till_done()
+
+    assert "Set ['John Doe'] as at home" in caplog.text
```

Comment:
Can this test some access to the pyatmo API instead of a log?

Reply:
So far there is no representation/state of persons known to Netatmo in HA.